### ISSUE TITLE
Unable to Replicate the results for BERT-based models

### ISSUE THREAD

**Issue by shirinssalehi** [thunlp/OpenMatch#19]
Hi!
I am trying to replicate the results for the transformer models. While the BERT-base model is trained properly, the Electra-base, Electra-large, and Roberta-large models are not being trained properly. I get MRRs below 0.25 which is lower than the numbers reported in the repo. In addition, I use the default parameters for the training. Is there any specific point I have to observe in order to get the models trained correctly?

**Comment by zkt12** [comment]
Hi,
Could you please show your scripts for Electra-base, Electra-large, and Roberta-large here?

Kaitao

**Comment by shirinssalehi** [comment]
Hello Toby!

Hope you are keeping safe!
Thank you very much for your response. I use the main train script provided
in the repository (train.py) for the ranking task. Are the results provided
in the repository for the ranking task or the classification task?
In addition, I set these arguments (-task ranking -model bert -vocab
google/electra-large-discriminator -pretrain
google/electra-large-discriminator -batch_size 16 -max_input 12800000
-epoch 3).
I have also attached the training script to this email.


Sincerely,
Shirin Seyedsalehi

**Comment by shirinssalehi** [comment]
This is the training procedure of Roberta-large.
It seems that something is wrong! The training loss does not follow a reasonable reduction pattern. After 18000 iterations the training loss starts increasing!
![image](https://user-images.githubusercontent.com/39522669/119398329-a9ae2c80-bcec-11eb-9456-bcba730e5582.png)

**Comment by zkt12** [comment]
> This is the training procedure of Roberta-large.
> It seems that something is wrong! The training loss does not follow a reasonable reduction pattern. After 18000 iterations the training loss starts increasing!
> ![image](https://user-images.githubusercontent.com/39522669/119398329-a9ae2c80-bcec-11eb-9456-bcba730e5582.png)

I will check this. Does Electra-large training have the same loss problem?

**Comment by zkt12** [comment]
and what's the learning rate and warmup step of Roberta-large?

**Comment by shirinssalehi** [comment]
Yes, Roberta-large has the same problem. I use the default parameters for
the learning rate and warm-up steps, which are 2e-5 and 1000 respectively.
And could you please let me know if the reported results on the open match
Github are for the classification task or the ranking task? because I use
the script for the ranking task.

**Comment by zkt12** [comment]
The reported results for MS MARCO passage ranking are for the ranking task. However, the recommended learning rate for Roberta-large is 5e-7, and 5e-6 for Electra-large, which are noted at the training part of https://github.com/thunlp/OpenMatch/blob/master/docs/experiments-msmarco.md. I think this is the problem :)
